server.go

```diff
@@ -501,9 +483,6 @@ func (s *Server) shutdown() {
 		return true
 	})
 
-	// wait for consumers to actually close
-	s.managerWg.Wait()
```

Comment:
Also note that `managerWg.Wait()` **together** with `manager.StopAcceptingConsumers()` ensures that no new consumer is **created** during the consumer shutdown.

Reply:
Right! However, I'm oriented to handling this scenario a bit differently. Currently, during a server's shutdown, we accept connections from new `Clients` and apparently new `Producers` while we don't accept new `Consumers` as you already mentioned.

Since there's no valid reason to create new clients and/or producers which will be closed some seconds/milliseconds later, we should modify this logic to first stop accepting new Client connections, before we start closing the registered consumers and producers.

Reply:
> Currently, during a server's shutdown, we accept connections from new `Clients` and apparently new `Producers` while we don't accept new `Consumers` as you already mentioned.

Correct. This is intentional, see https://github.com/skroutz/rafka/commit/af07cbf443789e59f2b9fbf495370dd7ad811139.

Reply:
Not directly related to the current PR but since it's been discussed, I think that having different shutdown policies complicates things and might lead to unflushed messages.
In the case we allow a producer to register itself and produce a high number of messages during our shutdown process we will end up [flushing a number of messages while closing](https://github.com/skroutz/rafka/blob/master/producer.go#L78) leaving a high probability of unflushed messages. Those unflushed messages won't be communicated to the client ( as is the case in rafka in general ) and due to the shutdown they won't event get reported since `STATS` will be overridden. That said, imo letting producers late-register might lead to unnoticed problems ( deficient monitoring is partly to blame here too )

Reply:
Isn't this case also possible given a "unified policy" flow? For example, assuming the flow:

1. stop accepting new clients
2. close consumers & producers
3. shutdown

In the above case, we can run in the same problem you mentioned, if a producer registers right before (1) and produces a large number of messages.

The possibility of unflushed messages will always be there IMHO and is irrelevant to rafka's shutdown process. That said, we should implement monitoring and handling of unflushed messages. Also #35 is something that should help.

Reply:
Of course this can happen in any flow.
The difference is that now we allow more time for producers to register and the window for unflushed messages just gets bigger.

Regarding #35, for producers that need to make sure that all their messages are flushed [Producer.Flush](https://github.com/skroutz/rafka/blob/f9311550d7ec82bf7ddec7e5094d34912780bf60/producer.go#L70) can be used.

In any case, this is out of the current PR's scope and I was just expressing some concerns on the subject. We'll revisit this when it's time comes.

Reply:
Let's not mess things up! This is part of another PR/issue/etc.